src/uwtools/drivers/driver.py

```diff
@@ -7,25 +7,30 @@
 
 from uwtools import config_validator
 from uwtools.logger import Logger
+from uwtools.scheduler import BatchScript
 
 
 class Driver(ABC):
     """
     An abstract class representing drivers for various NWP tools.
     """
 
-    def __init__(self, config_file: str, log: Optional[Logger] = None):
+    def __init__(
+        self, config_file: str, log: Optional[Logger] = None, dry_run: Optional[bool] = False
```

Comment:
Unless we need to be able to explicitly pass in a `None` value for `dry_run`, which doesn't immediately make sense to me, `dry_run: Optional[bool] = False` should just be `dry_run: bool = False`, which means that the argument is optional but will be `False` if not supplied. `Optional[<type>]` is shorthand for `Union[<type>, None]`, meaning that a value is expected, but that value may be `None`.

If I'm not mistaken here and this can be simplified, I think the line
```
self._dry_run = dry_run if dry_run is not None else False
```
can be just
```
self.dry_run = dry_run
```
since the `dry_run` argument will definitely have a value of either `True` or `False`.

Reply:
OK, good catch - I have a bad habit of over-stating things to a fault, I'll clean it up, thanks!